src/tracked_struct.rs

```diff
@@ -368,7 +381,7 @@ where
 
         let identity_hash = IdentityHash {
             ingredient_index: self.ingredient_index,
-            hash: crate::hash::hash(&C::id_fields(&fields)),
+            hash: crate::hash::hash(&C::untracked_fields(&fields)),
```

Comment:
Note that `IdentityHash`'s docs state

> This includes the ingredient index of that struct type plus the hash of its id fields.

So if I see this right we have flipped the fields that we hash here now right? before we hashed  the tracked fields, not hash the untracked fields which sounds wrong to me?

Reply:
Flipping here is correct. We should have updated the documentation. 

The identity of a tracked struct is now defined by all untracked fields. If any untracked field changes, then we just assume it's a different tracked struct. This "trick" ensures that we get away without tracking dependencies on individual fields because the id from rev `a` and rev `b` now no longer compare equal

Reply:
Yeah "id fields" have effectively been renamed to "untracked fields".